--- /dev/null	2022-12-31 10:33:36
+++ WORKSPACE	2022-12-31 10:31:42
@@ -0,0 +1,10 @@
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository", "new_git_repository")
+
+# Bazel Skylib rules.
+git_repository(
+    name = "bazel_skylib",
+    tag = "1.2.1",
+    remote = "https://github.com/bazelbuild/bazel-skylib.git",
+)
+load("@bazel_skylib//:workspace.bzl", "bazel_skylib_workspace")
+bazel_skylib_workspace()
