diff --git a/src/scip/buildflags.c b/src/scip/buildflags.c
index 4fbbc0e3b..3dc550347 100644
--- src/scip/buildflags.c
+++ src/scip/buildflags.c
@@ -0,0 +1,2 @@
+#define SCIP_BUILDFLAGS " ARCH=x86_64\n COMP=gnu\n DEBUGSOL=false\n EXPRINT=none\n GAMS=false\n SYM=bliss\n GMP=false\n IPOPT=false\n IPOPTOPT=opt\n WORHP=false\n WORHPOPT=opt\n LPS=spx2\n LPSCHECK=false\n LPSOPT=opt\n NOBLKBUFMEM=false\n NOBLKMEM=false\n NOBUFMEM=false\n OPT=opt\n OSTYPE=linux\n PARASCIP=true\n READLINE=false\n SANITIZE=\n SHARED=false\n USRARFLAGS=\n USRCFLAGS=-fPIC\n USRCXXFLAGS=-fPIC\n USRDFLAGS=\n USRFLAGS=\n USRLDFLAGS=\n USROFLAGS=\n VERSION=7.0.1\n ZIMPL=false\n ZIMPLOPT=opt\n ZLIB=true"
+
diff --git a/src/scip/githash.c b/src/scip/githash.c
new file mode 100644
index 000000000..660dc1976
--- /dev/null
+++ src/scip/githash.c
@@ -0,0 +1 @@
+#define SCIP_GITHASH "b41d526acf"
diff --git a/src/symmetry/compute_symmetry_bliss.cpp b/src/symmetry/compute_symmetry_bliss.cpp
index db734cb18..2fda08fd4 100644
--- src/symmetry/compute_symmetry_bliss.cpp
+++ src/symmetry/compute_symmetry_bliss.cpp
@@ -24,8 +24,8 @@
 #include "compute_symmetry.h"

 /* include bliss graph */
-#include <bliss/defs.hh>
-#include <bliss/graph.hh>
+#include <bliss-0.73/defs.hh>
+#include <bliss-0.73/graph.hh>

 #include <vector>
 #include <list>
