diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d81502..f29326d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.1)
 
 project(cctz)
 
@@ -58,7 +58,13 @@ function(cctz_target_set_cxx_standard target)
   endif()
 endfunction()
 
-add_library(cctz
+set(_HDRS
+  include/cctz/time_zone.h
+  include/cctz/civil_time_detail.h
+  include/cctz/zone_info_source.h
+  include/cctz/civil_time.h
+  )
+set(_SRCS
   src/civil_time_detail.cc
   src/time_zone_fixed.cc
   src/time_zone_fixed.h
@@ -77,11 +83,17 @@ add_library(cctz
   src/tzfile.h
   src/zone_info_source.cc
   )
+add_library(cctz ${_SRCS} ${_HDRS})
 cctz_target_set_cxx_standard(cctz)
+target_include_directories(cctz PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+  )
+set_target_properties(cctz PROPERTIES
+  PUBLIC_HEADER "${_HDRS}"
+  )
 add_library(cctz::cctz ALIAS cctz)
 
-target_include_directories(cctz PUBLIC include)
-
 if (BUILD_TOOLS)
   add_executable(time_tool src/time_tool.cc)
   cctz_target_set_cxx_standard(time_tool)
@@ -137,4 +149,27 @@ if (BUILD_TESTING)
   target_link_libraries(cctz_benchmark cctz::cctz benchmark::benchmark_main)
 endif()
 
+# Install
+include(GNUInstallDirs)
+install(TARGETS cctz
+  EXPORT ${PROJECT_NAME}-targets
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/cctz
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  )
+if (BUILD_TOOLS)
+  install(TARGETS time_tool
+    EXPORT ${PROJECT_NAME}-targets
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    )
+endif()
+set(CMAKE_INSTALL_CONFIGDIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+install(EXPORT ${PROJECT_NAME}-targets
+  NAMESPACE cctz::
+  DESTINATION ${CMAKE_INSTALL_CONFIGDIR}
+  )
+install(FILES cmake/${PROJECT_NAME}-config.cmake
+  DESTINATION ${CMAKE_INSTALL_CONFIGDIR}
+  )
+
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
