From 00663891e82c32fc032ee93ab8718e32a7a643d2 Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.ambasta@delhivery.com>
Date: Wed, 7 Jun 2017 19:37:49 +0530
Subject: [PATCH] Patch to fix invalid includedir for try_compile

---
 cmake/DetermineGflagsNamespace.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/DetermineGflagsNamespace.cmake b/cmake/DetermineGflagsNamespace.cmake
index e5e50cc..3a30df8 100755
--- a/cmake/DetermineGflagsNamespace.cmake
+++ b/cmake/DetermineGflagsNamespace.cmake
@@ -35,7 +35,7 @@ int main(int argc, char**argv)
         "${CMAKE_BINARY_DIR}" "${_check_file}"
         COMPILE_DEFINITIONS "${CMAKE_REQUIRED_DEFINITIONS}" -DGFLAGS_NAMESPACE=${_namespace}
         LINK_LIBRARIES "${gflags_LIBRARIES}"
-        CMAKE_FLAGS -DINCLUDE_DIRECTORIES:STRING="${gflags_INCLUDE_DIR}"
+        CMAKE_FLAGS "-DINCLUDE_DIRECTORIES:STRING=${gflags_INCLUDE_DIR}"
         OUTPUT_VARIABLE OUTPUT)
 
       if (${VARIABLE})
-- 
2.13.0

