diff --git a/CMakeLists.txt b/CMakeLists.txt
index 22b0048cd..e8beb5db0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,7 +50,7 @@ endif()
 
 option(PYTHON_BUILD_SETUP "Build Python interface from setup.py" OFF)
 message(STATUS "Build Python: ${PYTHON_BUILD_SETUP}")
-if (PYTHON_BUILD_SETUP) 
+if (PYTHON_BUILD_SETUP)
   set(BUILD_CXX OFF)
   set(BUILD_TESTING OFF)
 endif()
@@ -64,7 +64,7 @@ message(STATUS "Build all tests: ${ALL_TESTS}")
 
 option(ZLIB "ZLIB" ON)
 message(STATUS "ZLIB: ${ZLIB}")
-if (PYTHON_BUILD_SETUP) 
+if (PYTHON_BUILD_SETUP)
   set(ZLIB OFF)
 endif()
 
@@ -181,7 +181,7 @@ if (BUILD_CXX)
     endif()
   endif()
 
-endif() 
+endif()
 
 include(CheckCXXSourceCompiles)
 check_cxx_source_compiles(
@@ -297,7 +297,7 @@ if(MSVC)
   add_compile_options("$<$<COMPILE_LANGUAGE:C,CXX>:-D_CRT_SECURE_NO_WARNINGS>")
   add_compile_options("$<$<COMPILE_LANGUAGE:C,CXX>:/MP>")
 
-  # Try to split large pdb files into objects. 
+  # Try to split large pdb files into objects.
   # https://github.com/tensorflow/tensorflow/issues/31610
   add_compile_options("/Z7")
   add_link_options("/DEBUG:FASTLINK")
@@ -353,7 +353,7 @@ endif()
 # set (CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address -fsanitize=undefined")
 
 # if zlib is found, then we can enable reading zlib-compressed input
-if(ZLIB)
+if(ZLIB AND NOT TARGET ZLIB::ZLIB)
   find_package(ZLIB 1.2.3)
 endif()
 
@@ -455,7 +455,7 @@ if(NOT FAST_BUILD)
         COMMENT "Resetting code coverage counters to zero.")
     endif()
   endif()
- 
+
   include_directories(
     ${HIGHS_BINARY_DIR}
     ${HIGHS_SOURCE_DIR}/app
@@ -496,7 +496,7 @@ if(NOT FAST_BUILD)
 
 else(FAST_BUILD)
   message(STATUS "FAST_BUILD set to on.")
-  
+
   option(BUILD_EXAMPLES "Build examples" ON)
   message(STATUS "Build examples: ${BUILD_EXAMPLES}")
 
diff --git a/cmake/sources.cmake b/cmake/sources.cmake
index cdf109699..e9e1362af 100644
--- a/cmake/sources.cmake
+++ b/cmake/sources.cmake
@@ -128,7 +128,7 @@ set(ipx_sources
   ipm/ipx/symbolic_invert.cc
   ipm/ipx/timer.cc
   ipm/ipx/utils.cc)
-  
+
 set(ipx_headers
   ipm/ipx/basiclu_kernel.h
   ipm/ipx/basiclu_wrapper.h
@@ -186,6 +186,7 @@ set(highs_sources
     lp_data/HighsDebug.cpp
     lp_data/HighsInfo.cpp
     lp_data/HighsInfoDebug.cpp
+    lp_data/HighsDeprecated.cpp
     lp_data/HighsInterface.cpp
     lp_data/HighsLp.cpp
     lp_data/HighsLpUtils.cpp
