diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 71a0f37a..bbb749d3 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -40,7 +40,7 @@ else (BUILD_SHARED_LIBS)
 endif (BUILD_SHARED_LIBS)
 option(protobuf_BUILD_SHARED_LIBS "Build Shared Libraries" ${protobuf_BUILD_SHARED_LIBS_DEFAULT})
 include(CMakeDependentOption)
-cmake_dependent_option(protobuf_MSVC_STATIC_RUNTIME "Link static runtime libraries" ON
+cmake_dependent_option(protobuf_MSVC_STATIC_RUNTIME "Link static runtime libraries" OFF
   "NOT protobuf_BUILD_SHARED_LIBS" OFF)
 set(protobuf_WITH_ZLIB_DEFAULT ON)
 option(protobuf_WITH_ZLIB "Build with zlib support" ${protobuf_WITH_ZLIB_DEFAULT})
diff --git a/cmake/install.cmake b/cmake/install.cmake
index 9b2ae93c..f13caa5d 100644
--- a/cmake/install.cmake
+++ b/cmake/install.cmake
@@ -118,19 +118,20 @@ configure_file(protobuf-module.cmake.in
 configure_file(protobuf-options.cmake
   ${CMAKE_INSTALL_CMAKEDIR}/protobuf-options.cmake @ONLY)
 
-# Allows the build directory to be used as a find directory.
-
-if (protobuf_BUILD_PROTOC_BINARIES)
-  export(TARGETS libprotobuf-lite libprotobuf libprotoc protoc
-    NAMESPACE protobuf::
-    FILE ${CMAKE_INSTALL_CMAKEDIR}/protobuf-targets.cmake
-  )
-else (protobuf_BUILD_PROTOC_BINARIES)
-  export(TARGETS libprotobuf-lite libprotobuf
-    NAMESPACE protobuf::
-    FILE ${CMAKE_INSTALL_CMAKEDIR}/protobuf-targets.cmake
-  )
-endif (protobuf_BUILD_PROTOC_BINARIES)
+if ("^${CMAKE_SOURCE_DIR}$" STREQUAL "^${PROJECT_SOURCE_DIR}$")
+  # Allows the build directory to be used as a find directory.
+  if (protobuf_BUILD_PROTOC_BINARIES)
+    export(TARGETS libprotobuf-lite libprotobuf libprotoc protoc
+      NAMESPACE protobuf::
+      FILE ${CMAKE_INSTALL_CMAKEDIR}/protobuf-targets.cmake
+    )
+  else (protobuf_BUILD_PROTOC_BINARIES)
+    export(TARGETS libprotobuf-lite libprotobuf
+      NAMESPACE protobuf::
+      FILE ${CMAKE_INSTALL_CMAKEDIR}/protobuf-targets.cmake
+    )
+  endif (protobuf_BUILD_PROTOC_BINARIES)
+endif()
 
 install(EXPORT protobuf-targets
   DESTINATION "${CMAKE_INSTALL_CMAKEDIR}"
