diff --git a/build_defs.bzl b/build_defs.bzl
index 1e55548..6763bc6 100644
--- a/build_defs.bzl
+++ b/build_defs.bzl
@@ -41,6 +41,10 @@ def pybind_extension(

     native.cc_binary(
         name = name + ".so",
+        target_compatible_with = select({
+          "@platforms//os:windows": ["@platforms//:incompatible"],
+          "//conditions:default": [],
+        }),
         copts = copts + PYBIND_COPTS + select({
             "@pybind11//:msvc_compiler": [],
             "//conditions:default": [
@@ -59,6 +75,33 @@ def pybind_extension(
         **kwargs
     )
 
+    native.cc_binary(
+        name = name + ".dll",
+        target_compatible_with = select({
+          "@platforms//os:windows": [],
+          "//conditions:default": ["@platforms//:incompatible"],
+        }),
+        copts = copts + PYBIND_COPTS,
+        features = features + PYBIND_FEATURES,
+        linkopts = linkopts,
+        linkshared = 1,
+        tags = tags,
+        deps = deps + PYBIND_DEPS,
+        **kwargs
+    )
+
+    native.genrule(
+        name = name + "_pyd",
+        target_compatible_with = select({
+          "@platforms//os:windows": [],
+          "//conditions:default": ["@platforms//:incompatible"],
+        }),
+        srcs = [name + ".dll"],
+        outs = [name + ".pyd"],
+        cmd = "cp $< $@"
+    )
+
+
 # Builds a pybind11 compatible library. This can be linked to a pybind_extension.
 def pybind_library(
         name,
--
2.36.0.512.ge40c2bad7a-goog

