diff --git a/pybind11_protobuf/proto_cast_util.cc b/pybind11_protobuf/proto_cast_util.cc
index 5ca857b..8b00a4e 100644
--- a/pybind11_protobuf/proto_cast_util.cc
+++ b/pybind11_protobuf/proto_cast_util.cc
@@ -355,7 +355,7 @@ class PythonDescriptorPoolWrapper {
     // the Python DescriptorPool.
 
     // Find a file by file name.
-    bool FindFileByName(const std::string& filename,
+    bool FindFileByName(StringViewArg filename,
                         FileDescriptorProto* output) override {
       try {
         auto file = pool_.attr("FindFileByName")(filename);
@@ -371,7 +371,7 @@ class PythonDescriptorPoolWrapper {
     }
 
     // Find the file that declares the given fully-qualified symbol name.
-    bool FindFileContainingSymbol(const std::string& symbol_name,
+    bool FindFileContainingSymbol(StringViewArg symbol_name,
                                   FileDescriptorProto* output) override {
       try {
         auto file = pool_.attr("FindFileContainingSymbol")(symbol_name);
@@ -389,7 +389,7 @@ class PythonDescriptorPoolWrapper {
 
     // Find the file which defines an extension extending the given message type
     // with the given field number.
-    bool FindFileContainingExtension(const std::string& containing_type,
+    bool FindFileContainingExtension(StringViewArg containing_type,
                                      int field_number,
                                      FileDescriptorProto* output) override {
       try {
