diff --git a/java/private/junit5.bzl b/java/private/junit5.bzl
index 94018e2..a3415c5 100644
--- a/java/private/junit5.bzl
+++ b/java/private/junit5.bzl
@@ -90,9 +90,6 @@ def java_junit5_test(
         jvm_flags = jvm_flags + ["-DJUNIT5_EXCLUDE_ENGINES=%s" % ",".join(exclude_engines)]
 
     security_manager_flag_seen = False
-    for flag in jvm_flags:
-        if flag.startswith("-Djava.security.manager="):
-            security_manager_flag_seen = True
     if not security_manager_flag_seen:
         jvm_flags = jvm_flags + ["-Djava.security.manager=allow"]
 
