From e168f01d1bc1dd3d3c688d55132e5ef0d49be5bc Mon Sep 17 00:00:00 2001
From: Eric D <eric.dumont@eurodecision.com>
Date: Fri, 3 May 2019 15:26:46 +0200
Subject: [PATCH] Force building dependencies in parallel (Unix : -j6)

---
 cmake/utils.cmake | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/cmake/utils.cmake b/cmake/utils.cmake
index 0ae4eb461..b1ede8f8f 100644
--- a/cmake/utils.cmake
+++ b/cmake/utils.cmake
@@ -114,8 +114,11 @@ function(build_git_dependency)
     message(FATAL_ERROR "CMake step for ${GIT_DEP_NAME} failed: ${result}")
   endif()
 
+  if (NOT MSVC)
+	set(Unix_specific -- -j6)
+  endif()
   execute_process(
-    COMMAND ${CMAKE_COMMAND} --build project_build --config ${CMAKE_BUILD_TYPE}
+    COMMAND ${CMAKE_COMMAND} --build project_build --config ${CMAKE_BUILD_TYPE} ${Unix_specific}
     RESULT_VARIABLE result
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${GIT_DEP_NAME})
   if(result)
-- 
2.19.0.windows.1

